internal/datanode/data_node.go

```diff
@@ -944,7 +943,7 @@ func (node *DataNode) SyncSegments(ctx context.Context, req *datapb.SyncSegments
 		numRows:      req.GetNumOfRows(),
 	}
 
-	err = channel.InitPKstats(ctx, targetSeg, req.GetStatsLogs(), tsoutil.GetCurrentTime())
+	err := channel.InitPKstats(ctx, targetSeg, req.GetStatsLogs(), tsoutil.GetCurrentTime())
```

Comment:
If we have invalid segment in compactFrom, PK stats are still generated here?

Reply:
The information related to compactFrom is not involved. The bloomFilterFile is mainly read according to the path about the compactTo, so repeating init should be no problem.